remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/BaseRemoteQueryManager.java

```diff
@@ -5,38 +5,57 @@
 
 import org.infinispan.AdvancedCache;
 import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.logging.LogFactory;
 import org.infinispan.encoding.DataConversion;
+import org.infinispan.encoding.impl.StorageConfigurationManager;
+import org.infinispan.factories.ComponentRegistry;
 import org.infinispan.factories.annotations.Inject;
 import org.infinispan.factories.scopes.Scope;
 import org.infinispan.factories.scopes.Scopes;
 import org.infinispan.marshall.core.EncoderRegistry;
 import org.infinispan.query.dsl.IndexedQueryMode;
 import org.infinispan.query.dsl.Query;
 import org.infinispan.query.remote.client.impl.QueryRequest;
+import org.infinispan.query.remote.impl.logging.Log;
 
 /**
  * @since 9.4
  */
 @Scope(Scopes.NAMED_CACHE)
 abstract class BaseRemoteQueryManager implements RemoteQueryManager {
 
+   private static final Log log = LogFactory.getLog(BaseRemoteQueryManager.class, Log.class);
+
    final AdvancedCache<?, ?> cache;
    private final QuerySerializers querySerializers;
    private final DataConversion keyDataConversion;
    private final DataConversion valueDataConversion;
+   private final boolean cacheQueryable;
+   private final MediaType storageType;
+   private final boolean unknownMediaType;
 
    @Inject protected EncoderRegistry encoderRegistry;
 
-   BaseRemoteQueryManager(AdvancedCache<?, ?> cache, QuerySerializers querySerializers) {
+   BaseRemoteQueryManager(AdvancedCache<?, ?> cache, QuerySerializers querySerializers, ComponentRegistry cr) {
       this.cache = cache;
       this.querySerializers = querySerializers;
       this.keyDataConversion = cache.getKeyDataConversion();
       this.valueDataConversion = cache.getValueDataConversion();
+      StorageConfigurationManager storageConfigurationManager = cr.getComponent(StorageConfigurationManager.class);
+      this.storageType = storageConfigurationManager.getValueStorageMediaType();
+      this.cacheQueryable = storageConfigurationManager.isQueryable();
+      this.unknownMediaType = storageType.match(MediaType.APPLICATION_UNKNOWN);
    }
 
    @Override
    public byte[] executeQuery(String queryString, Map<String, Object> namedParametersMap, Integer offset, Integer maxResults,
                               IndexedQueryMode queryMode, AdvancedCache<?, ?> cache, MediaType outputFormat) {
+      if (unknownMediaType) {
+         log.warnNoMediaType(cache.getName());
```

Comment:
So unknown will trigger a warning not an error. But will it actually work?

Reply:
I presume it will work. But under what assuptions? Will it try to use it as protobuf event though it is not known to be ?

Reply:
That is the reason for the WARN. If someone creates a server cache and don't specify any encoding at all, but at runtime protobuf is stored, it will work. If garbage is stored, it will not, but at least it will print a WARN.

Reply:
Ok. I have a faint memory of the existence of a jira about not allowing UNKNOWN anymore could you link it here for posterity please @gustavonalle ?

Reply:
We need to revive this discussion for ISPN 12

Reply:
sure, but do we have jira for it ?

Reply:
not yet, as there is no consensus on making mandatory the encoding config, or even make the default as protobuf